asm_to_pil/src/vm_to_constrained.rs

```diff
@@ -413,45 +419,41 @@ impl<T: FieldElement> ASMPILConverter<T> {
     fn handle_assignment(
         &mut self,
         _start: usize,
-        write_regs: Vec<String>,
-        assign_reg: Option<String>,
+        lhs_with_reg: Vec<(String, String)>,
         value: Expression<T>,
     ) -> CodeLine<T> {
-        assert!(write_regs.len() <= 1);
-        assert!(
-            assign_reg.is_some(),
-            "Implicit assign register not yet supported."
-        );
-        let assign_reg = assign_reg.unwrap();
+        assert!(lhs_with_reg.len() == 1);
+        let (write_regs, assign_reg) = lhs_with_reg.into_iter().next().unwrap();
         let value = self.process_assignment_value(value);
         CodeLine {
-            write_regs: [(assign_reg.clone(), write_regs)].into_iter().collect(),
+            write_regs: [(assign_reg.clone(), vec![write_regs])]
+                .into_iter()
+                .collect(),
             value: [(assign_reg, value)].into(),
             ..Default::default()
         }
     }
 
     fn handle_functional_instruction(
         &mut self,
-        write_regs: Vec<String>,
-        assign_reg: String,
+        lhs_with_regs: Vec<(String, String)>,
         instr_name: String,
-        args: Vec<Expression<T>>,
+        mut args: Vec<Expression<T>>,
     ) -> CodeLine<T> {
-        assert!(write_regs.len() == 1);
         let instr = &self
             .instructions
             .get(&instr_name)
             .unwrap_or_else(|| panic!("Instruction not found: {instr_name}"));
-        assert_eq!(instr.outputs.len(), 1);
-        let output = instr.outputs[0].clone();
-        assert!(
-            output == assign_reg,
-            "The instruction {instr_name} uses the assignment register {output}, but the caller uses {assign_reg} to further process the value.",
-        );
+        let output = instr.outputs.clone();
+
+        for (o, (_, r)) in output.iter().zip(lhs_with_regs.iter()) {
+            assert!(
+                o == r,
+                "The instruction {instr_name} uses the output register {o}, but the caller uses {r} to further process the value.",
```

Comment:
can you also add a test that expects this

Reply:
I think actually this would already fail earlier, in `analysis/src/vm/inference.rs`, so this error is unreachable.

Reply:
I added a test `multi_return_wrong_assignment_registers.asm`. As mentioned, it hits another error message, but still good to test it!